Applications/SystemMonitor/main.cpp

```diff
@@ -364,5 +369,79 @@ GWidget* build_devices_tab()
     devices_table_view->set_model(GSortingProxyModel::create(DevicesModel::create()));
     devices_table_view->model()->update();
 
+    auto create_device = GAction::create("Create Device", [devices_table_view](const GAction&) {
+        if (getuid() != 0) {
+            GMessageBox::show("Only root can do that!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        auto model = devices_table_view->model();
+        String mount_point = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Device), GModel::Role::Sort).as_string();
+        unsigned major = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Major), GModel::Role::Sort).as_uint();
+        unsigned minor = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Minor), GModel::Role::Sort).as_uint();
+        String type = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Type), GModel::Role::Sort).to_string();
+
+        // FIXME: Paths seem to not be empty for a non-created device??? This is a nasty hack.
+        if (mount_point.starts_with("/")) {
+            GMessageBox::show("This device already exists!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        // Alright, the device doesn't exist so let's go!
+        GInputBox dev_path("Where should this device be created at?", "");
+        int dialog_return = dev_path.exec();
+        mount_point = dev_path.text_value();
+
+        // No assertions failed in THIS dojo!
+        if (dialog_return == GDialog::ExecResult::ExecCancel)
+            return;
+
+        if (mount_point.split('/').at(0) != "dev") {
+            GMessageBox::show("Devices must be mounted in /dev/!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        if (mount_point.is_empty()) {
+            GMessageBox::show("That's not a valid mount point!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        if (type.to_lowercase() == "block" || type.to_lowercase() == "character") {
+            // We have all the right info, so let's mount the device by calling 'mknod'
+            StringBuilder string_args;
+            string_args.append(mount_point);
+            string_args.append(" ");
+            string_args.append(type.to_lowercase() == "character" ? "c" : "b");
+            string_args.append(" ");
+            string_args.appendf("%d", major);
+            string_args.append(" ");
+            string_args.appendf("%d", minor);
+
+            Vector<String> args = string_args.to_string().split(' ');
```

Comment:
What's the point of first joining the args like that and then splitting them back? 😉

Reply:
Hrrm, I think I was originally going to pass the entire string to `exec` before I realised it took a `va_list` oops...